src/main/kotlin/ru/nsu/fevent/service/EventService.kt

```diff
@@ -17,4 +21,27 @@ class EventService(val eventRepository: EventRepository, val userRepository: Use
 
         return EventMapper.mapEntityToDto(savedEvent, UserMapper.mapEntityToDto(creator))
     }
+
+    fun viewEvents(s: String, sort: String, page: Int): List<EventViewDto>{
+        var direction = Sort.by(Sort.Direction.ASC, "name")
+        if (sort == "desc"){
+            direction = Sort.by(Sort.Direction.DESC, "name")
+        }
+
+        val perPage = 10
+        val searchedEvents = eventRepository.searchByName(s, PageRequest.of(page - 1, perPage, direction))
+
+        var viewEvents = ArrayList<EventViewDto>()
+        for (i in searchedEvents.indices) {
+            viewEvents += EventMapper.mapEntityToViewDto(searchedEvents[i])
+        }
```

Comment:
val searchedEvents = eventRepository.searchByName(s, PageRequest.of(page - 1, perPage, direction))
            .stream()
            .map { event -> EventMapper.mapEntityToViewDto(event) }
            .toList()